### ISSUE TITLE
Loading the weight file with use_weight_norm parameter using the updated compiled_tcn function

### ISSUE THREAD

**Issue by hlmhlr** [philipperemy/keras-tcn#282]
I used earlier version of this repo, trained the TCN model with following parameters, and installed libraries version are (tensorflow==2.11.1 and keras-tcn=3.5.0): 

```
model = compiled_tcn(
        return_sequences=False,
         num_feat=4, # 
        num_classes=args.num_classes,  # from CLI
        nb_filters=40,
        kernel_size=8,
        dilations=[2 ** i for i in range(9)],
        nb_stacks=1,
        max_len=x_train[0:1].shape[1],
        use_weight_norm=True,
        use_skip_connections=True,
        dropout_rate=0.05,
        opt=args.optimizer, # from CLI
        lr=args.lr
    )
```
For some reasons, I had to shift the code onto the colab notebook, where I tried to install the same version of tensorflow and keras-tcn, but it only allowed me to install the tensorflow version greater than or equal to 2.12 and the updated version of keras-tcn. 

Now, the latest repo of keras-tcn has removed the parameter `use_weight_norm`  from `compiled_tcn` function, and therefore, I am unable to load the weights which I saved with the model mentioned above, and thus to evaluate the data. 

How can I load the weight file (in which the `use_weight_norm` parameter was used) with the latest `compiled_tcn` mdel function from which  the  `use_weight_norm` parameter has been removed.

**Comment by philipperemy** [comment]
@hlmhlr on Colab you should be able to force install some deps:

```
pip uninstall -y tensorflow keras keras-tcn
pip install tensorflow==2.11.1 keras-tcn==3.5.0
```

Then reload the model exactly like before:

```
from tcn import compiled_tcn

model = compiled_tcn(
    return_sequences=False,
    num_feat=4,
    num_classes=num_classes,
    nb_filters=40,
    kernel_size=8,
    dilations=[2 ** i for i in range(9)],
    nb_stacks=1,
    max_len=max_len,
    use_weight_norm=True,
    use_skip_connections=True,
    dropout_rate=0.05,
    opt=optimizer,
    lr=lr
)

model.load_weights("weights.h5")
```

If you can't do that you can also load the model and discard those layers:

```
model.load_weights("weights.h5", by_name=True, skip_mismatch=True)
```

But:
- weight normalization layers will not load correctly
- results may differ.

So this is not ideal for research evaluation.
